OpenMarket/OpenMarket/Application/Presentation/View/GridCollectionCell.swift

```diff
@@ -0,0 +1,147 @@
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = UIFont.preferredFont(forTextStyle: .title3)
+        
+        return label
+    }()
+    
+    private let originalPriceLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = UIFont.preferredFont(forTextStyle: .caption1)
+        label.textColor = .gray
+        
+        return label
+    }()
+    
+    private let bargainPriceLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = UIFont.preferredFont(forTextStyle: .caption1)
+        label.textColor = .gray
+        
+        return label
+    }()
+    
+    private let leftoverLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = UIFont.preferredFont(forTextStyle: .caption1)
+        label.textColor = .gray
+        
+        return label
+    }()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        configureGridCell()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("not implemnted")
+    }
+    
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        
+        productImageView.image = nil
+        productNameLabel.text = nil
+        originalPriceLabel.text = nil
+        originalPriceLabel.textColor = .systemGray
+        bargainPriceLabel.text = nil
+        bargainPriceLabel.textColor = .systemGray
+        leftoverLabel.text = nil
+        leftoverLabel.textColor = .systemGray
+    }
+    
+}
+
+extension GridCollectionCell {
+    private func configureGridCell() {
+        contentView.addSubview(rootStackView)
+        rootStackView.addArrangedSubview(productImageView)
+        rootStackView.addArrangedSubview(labelStackView)
+        
+        labelStackView.addArrangedSubview(productNameLabel)
+        labelStackView.addArrangedSubview(originalPriceLabel)
+        labelStackView.addArrangedSubview(bargainPriceLabel)
+        labelStackView.addArrangedSubview(leftoverLabel)
+
+        let inset = CGFloat(10)
+            
+        NSLayoutConstraint.activate([
+            rootStackView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: inset),
+            rootStackView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -inset),
+            rootStackView.topAnchor.constraint(equalTo: contentView.topAnchor, constant: inset),
+            rootStackView.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -inset)
+        ])
+    }
+    
+    func updateUI(_ data: ProductEntity) {
+        productImageView.image = data.thumbnailImage
+        productNameLabel.text = data.name
+        originalPriceLabel.text = data.currency + " " + data.originalPrice.numberFormatter()
```

Comment:
```suggestion
        originalPriceLabel.text = viewModel.originalPriceText
```

뷰모델을 만들어 뷰모델을 넘기면 이런식으로 바로 할당 가능합니다. 뷰모델은 프레젠터(Presenter)로서의 역할을 합니다.
이 방법의 좋은 점은 
1) 뷰의 할당되는 값이 잘 나오는지 뷰모델의 메소드나 연산프로퍼티를 테스트하면 되므로 즉 뷰와의 의존성이 없으므로 테스트하기 쉽습니다.
2) 기획이 바뀌어서 originalPriceLabel.text의 값을 바꾸려 할 때 viewModel의 메소드(`originalPriceText`)만 바꾸면 된다. 
즉 여기 예시처럼 여러 커스텀 뷰에서의 코드를 일일이 바꾸지 않아도 되는 장점이 있습니다.